### DISCUSSION TITLE
Spacing to be used for `maxWidth` classes

### DISCUSSION THREAD

**Discussion by markmead** [tailwindlabs/tailwindcss#6146] [Ideas]
As it stands `spacing` is used for:

> padding, margin, width, height, maxHeight, gap, inset, space, and translate core plugins.

Could this include `maxWidth`? It seems odd that it doesn't.

As it stands I'm currently doing the following to make this work.

```js
extend: {
  spacing: {
    ...newSpacing,
  },
  maxWidth: (theme) => ({
    ...theme('spacing'),
  }),
},
```

This includes classes like `max-w-64` and any new classes I'm adding in the `newSpacing` object.

**Comment by adamwathan** [comment]
Main reason it’s not included is because we historically have had to be concerned about file size. I think we’ll probably add it for 3.0 but at the same time stuff like max-w-6 is pretty useless in real projects so it feels a bit weird to include it by default. Can you share some specific examples of where you’ve needed it?

**Reply by markmead** [reply]
I had a further think about this and came to the same conclusion. It will be silly to write `max-w-4`.

I think the only benefit of it being included is the example I gave but that doesn't apply to everyone.

Thanks for the response.

**Comment by Hodkiewicz** [comment]
Thanks for the response but the code error at the runtime .. i am still searching for the proper solution !!  [Tell The Bell](https://www.tellthebell.mobi/)

**Reply by markmead** [reply]
It works the way shown - https://play.tailwindcss.com/aB85ADtvpr?file=config
